Skip parsing parameters in single-line comments

rherrmann opened the pull request:
Extend the lexer grammars used to parse statements for colon
parameters, hash parameters, and attribute definitions to
recognize single-line comments starting with `--` and `//`.
As a result, placeholders within comments will remain
untouched when pre-processing statements.

https://github.com/jdbi/jdbi/issues/1712

leaumar commented on core/src/main/antlr4/org/jdbi/v3/core/internal/lexer/ColonStatementLexer.g4:
```diff
@@ -27,7 +27,7 @@ fragment NAME: JAVA_LETTER | [0-9] | '.' | '?.';
 fragment JAVA_LETTER : [a-zA-Z$_] | ~[\u0000-\u007F\uD800-\uDBFF] | [\uD800-\uDBFF] [\uDC00-\uDFFF];
 
-COMMENT: '/*' .*? '*/';
+COMMENT: '/*' .*? '*/' | '--' ~('\r' | '\n')* | '//' ~('\n' | '\r')*;
```
Is there any particular reasoning behind the different order of \\r and \\n in the two sides here, or just coincidence? If no reason, I'd prefer both in \\r\\n order (and the same in the other files of course)

rherrmann replied:
Hell no, thanks for catching this. I've also aligned newline patterns in `SqlScriptLexer.g4` to use \r\n order, which is otherwise unchanged.

leaumar reviewed (approved):
Looks neat to me!

leaumar commented:
There's no realistic and supportable situation where something that looks like an inline comment in a query is actually not an inline comment, right guys?

rherrmann commented:
> There's no realistic and supportable situation where something that looks like an inline comment in a query is actually not an inline comment, right guys?
> 
> Edit: perhaps add a test that e.g. string literals with comment markers inside them are still parsed as strings?

I've added a `testCommentCharsInQuotes` to the parameter parsing tests. For the `DefinedAttributeTemplateEngine`, I've added a test to ensure that comment chars within angle brackets aren't recognized as such. Not entirely sure though, if that makes sense. What do you think?

BTW: `TestColonPrefixSqlParser` contains two tests with similar names that are apparently testing the same thing: `testHashInColumnNameOkay` and `testHashInColumnNameWorks`. Shall I remove one of them - within this change or in a separate one?

leaumar commented on core/src/test/java/org/jdbi/v3/core/statement/TestColonPrefixSqlParser.java:
```diff
@@ -52,6 +52,13 @@ public void testEscapedQuoteInQuotes() {
 
+    @Test
+    public void testCommentCharsInQuotes() {
+        String sql = "select '-- // /* */'";
+        assertThat(parser.parse(sql, ctx))
+            .isEqualTo(ParsedSql.builder().append(sql).build());
```
nice, thanks

qualidafial commented:
Great test cases!

stevenschlansker reviewed (approved):
This is great, thank you all

stevenschlansker commented:
Duplicate test removed in 8648c1c56
